tests/test_timing.py

```diff
@@ -0,0 +1,87 @@
+import pytest
+import numpy as np
+import scipy.ndimage
+import torch
+
+from whisper.timing import dtw_cpu, dtw_cuda, median_filter
+
+
+sizes = [
+    (10, 20), (32, 16), (123, 1500), (234, 189),
+]
+shapes = [
+    (10,), (1, 15),  (4, 5, 345), (6, 12, 240, 512),
+]
+
+
+@pytest.mark.parametrize("N, M", sizes)
+def test_dtw(N: int, M: int):
```

Comment:
Was your reason to not use scipy.dtw licensing concerns or just speedup?

Reply:
dtw-python is GPL, as mentioned here -
https://github.com/openai/whisper/pull/869#issuecomment-1399100852